source/common/router/retry_state_impl.cc

```diff
@@ -240,6 +246,25 @@ RetryStatus RetryStateImpl::shouldRetry(bool would_retry, DoRetryCallback callba
   return RetryStatus::Yes;
 }
 
+bool RetryStateImpl::retryBudgetExceeded() {
+  if (!retry_budget_) {
+    return false;
+  }
+
+  // If a retry budget was configured, we cannot exceed the configured percentage of total
+  // outstanding requests/connections.
+  const uint64_t current_active = cluster_.resourceManager(priority_).connections().count() +
```

Comment:
This seems a bit iffy to me: today kinda this works before for H/1.1 and H/2 make use of either max_connections OR max_requests so this works, but there are requested features that might introduce max_connections to H/2 as well (https://github.com/envoyproxy/envoy/issues/7403). Envoy also doesn't reclaim H/1.1 connections until it needs to, so I think using `max_connections` to get a sense for the current concurrency isn't perfect. 

I think ideally we'd be counting # of requests for HTTP/1.1 and just do max_requests + max_pending, but I don't think we can just add that enforcement safely at this point (since users might have bumped max_con but not max_req). Maybe just note this limitation in the docs for now?

Reply:
Noted in the docs. Let me know if the latest patch communicates this well enough.